Blog: PyMC v4.0 release hackathon (#41)
pymc-devs/pymc.io: blog/dev_hackathon_v4_release.md

- Thomas Wiecki (Germany)
- Ricardo Vieira (Romania)
- Christian Luhmann (New York, USA)
- Alex Andorra (France)
- Michael Osthege (Germany)
- Oriol Abril Pla (Finland)
- Chris Fonnesbeck (Nashville, USA)
- Osvaldo Martin (Argentina)
- Maxim Kochurov (Moscow, Russia)
- Reshama Shaikh (New York, USA)

The PyMC project also have begun the 2022 [Google Summer of Code](https://github.com/pymc-devs/pymc/wiki/GSoC-2022-projects) (GSOC) program and the following GSOC folks joined:  

- Connor Hassan (Australia)
- Purna Chandra Mansingh (India)
- Danh Phan (Australia)
- Larry Dong (Canada)

A number of contributors joined from the community:  

- Fernando Irarrázaval
- Ivan Finaev
- percevalve
- Rob Zinkov (Germany)

There were lots of people throughout the week, and it is likely we are forgetting some folks. If we forgot you, let us know!

:::{figure-md} hackathon_screenshot

![grid screenshot of 9 people](../_static/hackathon_v4_release/tweet1_hackathon.png)

> Day 2 of PyMC v4.0 Developer Hackathon, June 2, 2022; Photo credit: [Thomas Wiecki](https://www.pymc-labs.io/team/thomas-wiecki/); (from left to right): Alex Andorra, Christian Luhmann, Connor Hassan , Maxime Kochurov, Oriol Abril Pla, Purna C, Ricardo Vieira, Rob Zinkov, Thomas Wiecki

Author: ```suggestion
Day 2 of PyMC v4.0 Developer Hackathon, June 2, 2022; Photo credit: [Thomas Wiecki](https://www.pymc-labs.io/team/thomas-wiecki/); (from left to right): Alex Andorra, Christian Luhmann, Connor Hassan , Maxime Kochurov, Oriol Abril Pla, Purna Chandra Mansingh, Ricardo Vieira, Rob Zinkov, Thomas Wiecki
```

<blockquote class="twitter-tweet"><p lang="en" dir="ltr">If you want to learn more about how <a href="https://twitter.com/hashtag/PyMC?src=hash&amp;ref_src=twsrc%5Etfw">#PyMC</a> 4.0 and Aesara work together, check out this new in-depth tutorial by <a href="https://twitter.com/RicardoV944?ref_src=twsrc%5Etfw">@RicardoV944</a> and <a href="https://twitter.com/juanitorduz?ref_src=twsrc%5Etfw">@juanitorduz</a> <a href="https://t.co/7Wk4UQkqXC">https://t.co/7Wk4UQkqXC</a> <a href="https://twitter.com/AesaraDevs?ref_src=twsrc%5Etfw">@AesaraDevs</a> <a href="https://twitter.com/pymc_devs?ref_src=twsrc%5Etfw">@pymc_devs</a> <a href="https://t.co/Ro9RsGaHrf">pic.twitter.com/Ro9RsGaHrf</a></p>&mdash; Thomas Wiecki (@twiecki) <a href="https://twitter.com/twiecki/status/1534115464967819265?ref_src=twsrc%5Etfw">June 7, 2022</a></blockquote> <script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>

<br>

### Plot Conditional Adjusted Prediction

> The new version also brings Plot Conditional Adjusted Predictions, if you are interested in contributing we are still lacking an example of this feature in our gallery.  https://bambinos.github.io/bambi/main/examples.html

Reviewer: ```suggestion
The new version also brings Plot Conditional Adjusted Predictions, if you are interested in contributing we are still lacking an example of this feature in our [example gallery](https://bambinos.github.io/bambi/main/examples.html)
```

<blockquote class="twitter-tweet"><p lang="en" dir="ltr">The new version also brings Plot Conditional Adjusted Predictions, if you are interested in contributing we are still lacking an example of this feature in our gallery. <a href="https://t.co/mXB8aXV6yN">https://t.co/mXB8aXV6yN</a> <a href="https://t.co/w9TmZnvYQd">https://t.co/w9TmZnvYQd</a> <a href="https://t.co/kOn5WQx0Zk">pic.twitter.com/kOn5WQx0Zk</a></p>&mdash; Osvaldo Martin (@aloctavodia) <a href="https://twitter.com/aloctavodia/status/1533875990992125955?ref_src=twsrc%5Etfw">June 6, 2022</a></blockquote> <script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>

<br>

## Summary

> Get started with v `4.0` by installing [PyMC via `conda`](https://www.pymc.io/blog/v4_announcement.html#installation). Read the full [PyMC v4.0 Release Announcement](https://www.pymc.io/blog/v4_announcement.html#installation) and [PyMC v4.0 Release Notes](https://github.com/pymc-devs/pymc/blob/main/RELEASE-NOTES.md).

Reviewer: ```suggestion
Get started with v `4.0` by installing {ref}``PyMC via `conda` <pymc:installation>``. Read the full {ref}`v4_announcement` and [PyMC v4.0 Release Notes](https://github.com/pymc-devs/pymc/blob/main/RELEASE-NOTES.md).
```

### Connect with PyMC

Connect with PyMC via:
- Website:  [pymc.io](https://www.pymc.io)
- Discourse: [discourse.pymc.io](https://discourse.pymc.io)
- YouTube: [PyMCDevelopers](https://www.youtube.com/c/PyMCDevelopers)
- Star GH repo:  [pymc-devs/pymc](https://github.com/pymc-devs/pymc)
- Join Meetup: [pymc-online-meetup](https://www.meetup.com/pymc-online-meetup/)
- Twitter: [@pymc_devs](https://twitter.com/pymc_devs)
- LinkedIn: [@pymc](https://www.linkedin.com/company/pymc/)

### Support PyMC

- [Donate to the the PyMC project](https://numfocus.org/donate-to-pymc)

### References

> - [PyMC v4.0 Release Announcement](https://www.pymc.io/blog/v4_announcement.html)

Reviewer: ```suggestion
- {ref}`v4_announcement`
```

> (sprint_testimonial)=

Reviewer: This can't be the same one we used for the sprint testimonial. This defines a target so we can then use `` {ref}`target` `` to generate a link that points to this page. It needs to be unique and specific to this file.

Author: Updated target

> (sprint_testimonial)=

Author: ```suggestion
(version_four_release)=
```

# Dev Sprint: Getting the PyMC v4.0 Release Out

:::{post} Jun 24, 2022
:tags: sprint, contributing
:category: news
:author: Reshama Shaikh, Thomas Wiecki
:image: 0
:exclude:

The PyMC team coordinated a development hackathon in May/June 2022 to complete updates and documentation in order to release [v4.0 of the PyMC library](https://twitter.com/pymc_devs/status/1533863780601446401).
:::

![v4 release social media card](../_static/hackathon_v4_release/v4release_card.png)

The [v4.0 release of the PyMC library](https://twitter.com/pymc_devs/status/1533863780601446401) was officially announced on June 6, 2022.  To achieve this milestone, the PyMC team planned a week-long online sprint beginning May 30, 2022.  The goal was to complete remaining issues on the v4.0 release.  But, *mainly* it was to finalize the documentation and update the website for PyMC.

There are 3 separate websites for [PyMC](https://www.pymc.io/welcome.html):  
- [Website and Blog](https://www.pymc.io/welcome.html)
- [Documentation (API)](https://www.pymc.io/projects/docs/en/stable/api.html)

> - [Examples](https://www.pymc.io/projects/examples/en/latest/gallery.html)

Author: @OriolAbril We do not have to include this.
The goal(s):  
- show that there was a significant update to the websites, given how much time was invested in it.  
- somehow write that in text.

> There are 3 separate websites for [PyMC](https://www.pymc.io/welcome.html):  
> - [Website and Blog](https://www.pymc.io/welcome.html)
> - [Documentation (API)](https://www.pymc.io/projects/docs/en/stable/api.html)
> - [Examples](https://www.pymc.io/projects/examples/en/latest/gallery.html)

Author: ```suggestion
```

Author: @OriolAbril 
Deleting the section on the website.  If you would like to add screenshots to explain the work to the website, feel free to do it.  Or the one sentence might be sufficient:
```txt
But, *mainly* it was to finalize the documentation and update the website for PyMC.
```

And all 3 websites had separate navigation bars and it was confusing for the users, maybe for the library maintainers too?

This Dev sprint was held on the new PyMC Discord server which easily enables sharing video, camera and screen share and provides an easy way to hop between “channels”.  

We shared the sprint events on our new [PyMC Online Meetup](https://www.meetup.com/pymc-online-meetup/events/286202420/) to involve experienced contributors in the community as well. We had a mix of core contributors and newcomers, which is a great combination, as it enables us to be productive, but also to foster the new generation of core developers. The following team members attended, from various locations around the world:  

- Thomas Wiecki (Germany)
- Ricardo Vieira (Romania)
- Christian Luhmann (New York, USA)
- Alex Andorra (France)
- Michael Osthege (Germany)
- Oriol Abril Pla (Finland)
- Chris Fonnesbeck (Nashville, USA)
- Osvaldo Martin (Argentina)

> - Rob Zinkov (Germany)

Reviewer: Rob is a recurrent contributor, at least for now. After some discussion we eventually agreed recurrent contributors would not be considered part of the pymc team.

> - Reshama Shaikh (New York, USA)

Reviewer: At the very least Max @ferrine is missing from the list. I only attended part of a couple of the sessions so there might be more people missing.

> - Reshama Shaikh (New York, USA)

Author: ```suggestion
- Maxim Kochurov (Moscow, Russia)
- Reshama Shaikh (New York, USA)
```

The PyMC project also have begun the 2022 [Google Summer of Code](https://github.com/pymc-devs/pymc/wiki/GSoC-2022-projects) (GSOC) program and the following GSOC folks joined:  

- Connor Hassan (Australia)
- Purna Chandra Mansingh (India)
- Danh Phan (Australia)
- Larry Dong (Canada)

A number of contributors joined from the community:  

- Fernando Irarrázaval
- Ivan Finaev
- percevalve

There were lots of people throughout the week, and it is likely we are forgetting some folks. If we forgot you, let us know!

> <figure>
>  <img src="../_static/hackathon_v4_release/tweet1_hackathon.png" alt="grid screenshot of 9 people" max-height="20%" /> 
>  <figcaption>
>  Day 2 of PyMC v4.0 Developer Hackathon, June 2, 2022; Photo credit: <a href="https://www.pymc-labs.io/team/thomas-wiecki/">Thomas Wiecki</a>; (from left to right): Alex Andorra, Christian Luhmann, x? , ?, Oriol Abril Pla, Purna C, Ricardo Vieira, ?Rob Zinkov, Thomas Wiecki
>  </figcaption>
> </figure>

Reviewer: ```suggestion
:::{figure-md} hackathon_screenshot

![grid screenshot of 9 people](../_static/hackathon_v4_release/tweet1_hackathon.png)

Day 2 of PyMC v4.0 Developer Hackathon, June 2, 2022; Photo credit: [Thomas Wiecki](https://www.pymc-labs.io/team/thomas-wiecki/); (from left to right): Alex Andorra, Christian Luhmann, Connor Hassan , Maxime Kochurov, Oriol Abril Pla, Purna C, Ricardo Vieira, Rob Zinkov, Thomas Wiecki
:::
```

If needed, we can also add [keyword arguments](https://sphinx-primer.readthedocs.io/en/latest/core/directives.html) to control the image and caption appeareance. AFAIK, both the kwargs accepted by the [figure](https://docutils.sourceforge.io/docs/ref/rst/directives.html#figure) and [image](https://docutils.sourceforge.io/docs/ref/rst/directives.html#image-options) directives are accepted.

<br>

## Summary [call to action]
[*** Thomas: add a line here encouraging people to upgrade to v4.0 + link to v4 installation instructions.]

### Connect with PyMC

Connect with PyMC via:
- Website:  [pymc.io](https://www.pymc.io)
- Discourse: [discourse.pymc.io](https://discourse.pymc.io)
- YouTube: [PyMCDevelopers](https://www.youtube.com/c/PyMCDevelopers)
- Star GH repo:  [pymc-devs/pymc](https://github.com/pymc-devs/pymc)
- Join Meetup: [pymc-online-meetup](https://www.meetup.com/pymc-online-meetup/)
- Twitter: [@pymc_devs](https://twitter.com/pymc_devs)
- LinkedIn: [@pymc](https://www.linkedin.com/company/pymc/)

### Support PyMC

> You can support the PyMC project by making a donation to the [PyMC on Open Collective](https://opencollective.com/pymc#category-CONTRIBUTE).

Reviewer: I think this should be https://numfocus.org/donate-to-pymc instead. But I'll let @twiecki and @fonnesbeck confirm.

Author: We can be consistent and use the NumFOCUS link.

> You can support the PyMC project by making a donation to the [PyMC on Open Collective](https://opencollective.com/pymc#category-CONTRIBUTE).

Author: ```suggestion
- [Donate to the the PyMC project](https://numfocus.org/donate-to-pymc)
```

- [Website and Blog](https://www.pymc.io/welcome.html)
- [Documentation (API)](https://www.pymc.io/projects/docs/en/stable/api.html)
- [Examples](https://www.pymc.io/projects/examples/en/latest/gallery.html)

And all 3 websites had separate navigation bars and it was confusing for the users, maybe for the library maintainers too?

This Dev sprint was held on the new PyMC Discord server which easily enables sharing video, camera and screen share and provides an easy way to hop between “channels”.  

We shared the sprint events on our new [PyMC Online Meetup](https://www.meetup.com/pymc-online-meetup/events/286202420/) to involve experienced contributors in the community as well. We had a mix of core contributors and newcomers, which is a great combination, as it enables us to be productive, but also to foster the new generation of core developers. The following team members attended, from various locations around the world:  

- Thomas Wiecki (Germany)
- Ricardo Vieira (Romania)
- Christian Luhmann (New York, USA)
- Alex Andorra (France)
- Michael Osthege (Germany)
- Oriol Abril Pla (Finland)
- Chris Fonnesbeck (Nashville, USA)
- Osvaldo Martin (Argentina)

> - Rob Zinkov (Germany)

Author: ```suggestion
```

- Maxim Kochurov (Moscow, Russia)
- Reshama Shaikh (New York, USA)

The PyMC project also have begun the 2022 [Google Summer of Code](https://github.com/pymc-devs/pymc/wiki/GSoC-2022-projects) (GSOC) program and the following GSOC folks joined:  

- Connor Hassan (Australia)
- Purna Chandra Mansingh (India)
- Danh Phan (Australia)
- Larry Dong (Canada)

A number of contributors joined from the community:  

- Fernando Irarrázaval
- Ivan Finaev

> - percevalve
> 

Author: ```suggestion
- percevalve
- Rob Zinkov (Germany)
```
